attest/untrusted/src/sim.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! A simulated implementation of DCAP quote generation.
+
+use crate::TargetInfoError;
+use mc_attest_core::QuoteError;
+use mc_attest_verifier::{IAS_SIM_SIGNING_CHAIN, IAS_SIM_SIGNING_KEY};
+use mc_rand::McRng;
+use mc_sgx_core_types::{Report, ReportBody, TargetInfo};
+use mc_sgx_dcap_sys_types::{sgx_ql_ecdsa_sig_data_t, sgx_quote3_t, sgx_quote_header_t};
+use mc_sgx_dcap_types::Quote3;
+use mc_sgx_types::sgx_report_body_t;
+use p256::ecdsa::{signature::Signer, Signature, SigningKey, VerifyingKey};
+use std::{mem, str::FromStr};
+
+pub struct SimQuotingEnclave;
+
+impl SimQuotingEnclave {
+    /// Get a quote based on the application enclave's `report`.
+    pub fn quote_report(report: &Report) -> Result<Quote3<Vec<u8>>, QuoteError> {
+        let cert_data = pck_cert_chain();
+
+        let sgx_quote = sgx_quote(report.body(), &cert_data);
+        let mut quote_bytes = c_struct_as_bytes(&sgx_quote).to_vec();
+
+        let mut rng = McRng::default();
+        let signing_key = SigningKey::random(&mut rng);
+        let signature = (&signing_key as &dyn Signer<Signature>).sign(&quote_bytes);
+
+        let ecdsa_sig = ecdsa_sig_data(signature, signing_key.verifying_key(), report.body());
```

Comment:
This is actually passing in the app report body as the qe report body. Since this is all fake, I'm thinking of keeping it this way until/if we determine we need to construct a fake QE report body.